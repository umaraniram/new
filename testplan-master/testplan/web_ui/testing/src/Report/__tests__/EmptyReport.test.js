/* Tests the EmptyReport component. */
import React from 'react';
import {shallow} from 'enzyme';

import EmptyReport from '../EmptyReport';

describe('EmptyReport', () => {
  it("shallow renders and matches snapshot", () => {
    const emptyReport = shallow(<EmptyReport />);
    expect(emptyReport).toMatchSnapshot();
  });

  it("renders a custom error message", () => {
    const errorReport = shallow(<EmptyReport message="418 I'm a teapot" />);
    expect(errorReport).toMatchSnapshot();
  });
});

