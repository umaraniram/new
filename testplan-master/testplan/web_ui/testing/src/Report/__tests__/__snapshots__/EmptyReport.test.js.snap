// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyReport renders a custom error message 1`] = `
<div
  className="emptyReport_3hmsj"
>
  <Toolbar
    extraButtons={
      Array [
        <TimeButton
          updateTimeDisplayCbk={[Function]}
        />,
      ]
    }
    filterBoxWidth="22em"
    handleNavFilter={[Function]}
    updateEmptyDisplayFunc={[Function]}
    updateFilterFunc={[Function]}
    updateTagsDisplayFunc={[Function]}
  />
  <Nav
    displayEmpty={true}
    displayTags={false}
    displayTime={false}
    report={null}
    saveAssertions={[Function]}
  />
  <Message
    left="22em"
    message="418 I'm a teapot"
  />
</div>
`;

exports[`EmptyReport shallow renders and matches snapshot 1`] = `
<div
  className="emptyReport_3hmsj"
>
  <Toolbar
    extraButtons={
      Array [
        <TimeButton
          updateTimeDisplayCbk={[Function]}
        />,
      ]
    }
    filterBoxWidth="22em"
    handleNavFilter={[Function]}
    updateEmptyDisplayFunc={[Function]}
    updateFilterFunc={[Function]}
    updateTagsDisplayFunc={[Function]}
  />
  <Nav
    displayEmpty={true}
    displayTags={false}
    displayTime={false}
    report={null}
    saveAssertions={[Function]}
  />
  <Message
    left="22em"
    message="404: Page Not Found"
  />
</div>
`;
